src/Common/src/Interop/Unix/System.Native/Interop.readdir.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        internal enum NodeType : short
+        {
+            Unknown         = 0,
+            NamedPipe       = 1,
+            CharacterDevice = 2,
+            Directory       = 4,
+            BlockDevice     = 6,
+            File            = 8,
+            SymLink         = 10,
+            Socket          = 12,
+            BsdWhiteout     = 14,
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal unsafe struct DirectoryEntry
+        {
+            internal ushort      reclen;
+            internal ulong       inode_number;
+            internal ulong       offset;
+            internal NodeType    inode_type;
+            internal fixed byte  name[1024];
```

Comment:
Previously each call to readdir would just give us back a pointer, which we could then index into at the right offset to interpret the string data.  But it seems like the way this new version is structured is going to result in needing to copy > 1K of data for each entry... that's a fair amount of copying going on.  Is there an alternate approach we could take that wouldn't involve so much copying?  Maybe this is a case where we choose to diverge slightly in the name of performance, e.g. by having the native function just give back a structure that includes just the data we need, namely the inode_type, the string ptr, and the length?  @nguerrera, thoughts?